indra/llwindow/llkeyboard.cpp

```diff
@@ -254,6 +259,11 @@ BOOL LLKeyboard::handleTranslatedKeyDown(KEY translated_key, U32 translated_mask
 	mKeyDown[translated_key] = TRUE;
 	mCurTranslatedKey = (KEY)translated_key;
 	handled = mCallbacks->handleTranslatedKeyDown(translated_key, translated_mask, repeated);
+    // TODO: capture unhandled keys and translate any mapped buttons to game controller
```

Comment:
Unless controller keys are identical to keybord keys or should go throught mTranslateKeyMap, this is a wrong spot for this.